### PR TITLE
iOS/Safari 8 onupgradeneeded-bugfix

### LINKED ISSUE #15: Feature: Safari 8 Support
Hello,

have you tested it on Safari?
I'm running an app successfully on Chrome and FF on Mac, but Safari is not ok, but not giving any error.

Thanks,
Mitch

### PR DESCRIPTION
**Pull request by JaZo**
This pr will resolve #15 and #41 by adding support for iOS/Safari 8.

### CODE REVIEW

File: src/angular-indexed-db.coffee [hunk 1]
```diff
@@ -141,7 +143,8 @@ angular.module('indexedDB', []).provider '$indexedDB', ->
 
 
     keyRangeForOptions = (options) ->
-      IDBKeyRange.bound(options.beginKey, options.endKey) if options.beginKey and options.endKey
+      return IDBKeyRange.bound(options.beginKey, options.endKey) if options.beginKey and options.endKey
```
Comment (bramski) [inline src/angular-indexed-db.coffee]: This change is totally ineffective.  If you look at the JS that was generated from this coffee you have pretty much done nothing, other than make the return case ```null``` if there's no options, and I think ```undefined``` is more normal.

**Comment by bramski** [comment]
Great!  Thanks for this.  Do address some of the comments.  My main question though is... how do you know that this works?  Ideally we'd be able to run this through the safari and iOS emulator.  I'm happy if you can have a test that runs the specs in safari that can only be run on OS X.

**Comment by bramski** [comment]
Another PR was just merged which worked with shimIndexedDB.  Is this PR still relevant?

**Comment by JaZo** [comment]
IndexedDB is supported on iOS 8, albeit buggy. In the way shimIndexedDB is currently implemented in this service, it will never be used on iOS 8, so this PR is still relevant. This PR solves some fixable iOS 8 quirks. I will have another look at this PR and your previous comments.

**Comment by alexey-sh** [comment]
How it's going? Could anybody resolve conflicts?

**Comment by JaZo** [comment]
I rebased/squashed/edited the commits and now it only contains a workaround for the onupgradeneeded-bug (https://bugs.webkit.org/show_bug.cgi?id=136888) and a simple fix for error messages. I do not have any tests for this, but the solution is used in other widely used packages. Therefore I can not guarantee that this works on iOS 8

**Comment by bramski** [comment]
Okay.  Merged.  Thanks!

**Pull request merged**